test/wait_for_blocks.go

```diff
@@ -28,6 +29,45 @@ func WaitForBlocks(ctx context.Context, delta int, chains ...ChainHeighter) erro
 	return eg.Wait()
 }
 
+// WaitForInSync blocks until all nodes have heights greater than or equal to the chain height.
+func WaitForInSync(ctx context.Context, chain ChainHeighter, nodes ...ChainHeighter) error {
+	if len(nodes) == 0 {
+		panic("missing nodes")
+	}
+InSyncLoop:
+	for {
```

Comment:
nit:

I try to avoid loop labels like `InSyncLoop` because they are essentially `GOTO` statements. 

In this case, a `for select` may clean up and eliminate the need for one. 

```go
for {
  select {
  case <-ctx.Done:
     return ctx.Err()
  default:
    if err := heightsInSync(...); err != nil {
       // maybe log error
        continue
    }
    return nil
}
```

I may also wrap the errgroup stuff in a private function so that I can simply return an `err` and use control flow based on that error.

Reply:
Also, a unit test would help ensure we don't have races.

Reply:
added